cdap-app-fabric/src/main/java/io/cdap/cdap/app/deploy/ProgramRunDispatcher.java

```diff
@@ -0,0 +1,30 @@
+/*
+ * Copyright © 2022 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.cdap.app.deploy;
+
+import io.cdap.cdap.app.runtime.ProgramController;
+import io.cdap.cdap.internal.app.deploy.pipeline.AppLaunchInfo;
+
+/**
+ * This interface is used for providing the abstraction for running programs either in app-fabric
```

Comment:
for executing program-run logic either in-memory or remotely (e.g., in a worker pod)